ucan/src/builder.rs

```diff
@@ -231,12 +231,26 @@ where
 
     /// Claim a capability by inheritance (from an authorizing proof) or
     /// implicitly by ownership of the resource by this UCAN's issuer
-    pub fn claiming_capability<S, A>(mut self, capability: &Capability<S, A>) -> Self
+    pub fn claiming_capability<S, A>(mut self, capability: &CapabilityView<S, A>) -> Self
     where
         S: Scope,
-        A: Action,
+        A: Ability,
     {
-        self.capabilities.push(CapabilityIpld::from(capability));
+        self.capabilities.push(Capability::from(capability));
+        self
+    }
+
+    /// Claim a capability by inheritance (from an authorizing proof) or
+    /// implicitly by ownership of the resource by this UCAN's issuer
+    pub fn claiming_capability_from_data(mut self, capability: &Capability) -> Self {
```

Comment:
These poorly-named methods that operate on `&Capabillity` can be removed, renamed, or replace `claiming_capability()` (my preference, using `into()` for CapabilitySemantics)

Reply:
I like it!